### PR TITLE
NIFI-515 - DeleteSQS and PutSQS should offer batch processing

### PR DESCRIPTION
**Pull request by pvillard31**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/sqs/PutSQS.java [hunk 1]
```diff
@@ -108,43 +108,57 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
...
 
         request.setEntries(entries);
 
         try {
             client.sendMessageBatch(request);
         } catch (final Exception e) {
-            getLogger().error("Failed to send messages to Amazon SQS due to {}; routing to failure", new Object[]{e});
-            flowFile = session.penalize(flowFile);
-            session.transfer(flowFile, REL_FAILURE);
+            getLogger().error("Failed to send {} messages to Amazon SQS due to {}; routing to failure", new Object[]{flowFiles.size(), e});
```
Comment (jzonthemtn) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/sqs/PutSQS.java]: There could be a mix of successful/failed messages in the `SendMessageBatchResult`. Does that impact how the session is transferred?

Comment (pvillard31) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/sqs/PutSQS.java]: You're right, I forgot that we have individual responses for each entry part of the request. Will update the PR. Thanks @jzonthemtn

**Comment by pvillard31** [comment]
@jzonthemtn - I just pushed a commit to address your comment. Basically, I parse the request results to split failures and success. I also added a unit test for that case.

**Comment by jzonthemtn** [comment]
@pvillard31 That sounds great!

**Comment by jzonthemtn** [comment]
@pvillard31 Wondering if there's still community interest in this one? It would be useful for me at least. I can help with the merge conflicts if there is.

**Comment by pvillard31** [comment]
Hey @jzonthemtn - sorry for the late answer, I've been busy lately. I just rebased this PR against master. If it still looks good to you, I'm sure we can find someone willing to merge it.

**Comment by joewitt** [comment]
Closing due to inactivity.

**Pull request closed without merging**